game_engine.py

```diff
@@ -172,13 +184,191 @@ def calculate_settlement_scores(self, player, current_board):
+        :return: the score * the weight based on the player strategy
+        """
+        score = 0
+        weight = 0
+        strategy = player.strategy
+        resources = player.resources
+
+        #  not enough resources
+        if resources["sheep"] == 0 or resources["stone"] == 0 or resources["wheat"] == 0:
+            return score
+
+        # enough resources, check strategy
+        if strategy == "development":
+            weight = 1
+        else:
+            weight = 0.65
+
+        # give score for not having enough knights for +2 points
+        if player.knights < 3:
+            score = score + 1
+
+        else:
+            cards_pulled = 25 - len(current_board.development_deck)
+            deck_ratio = len(current_board.development_deck) / 25
+
+            score = score + deck_ratio
+
+        return weight*score
+
+    def draw_development_card(self, current_board, player):
+        card = current_board.development_deck.pop(0)
+
+        if card == "knight":
+            player.knights = player.knights + 1
+            
+        elif card == "victory_point":
+            player.points = player.points + 1
+        
+        else:
+            pass
+
+        return current_board
     
     def calculate_trade_score(self, player, current_board):
-        return 0
+        """
+        Calculate the score for choosing a development card this turn
+        :param player: The player object who is deciding what to do
+        :param current_board: The current state of the board
+        :return: the score * the weight based on the player strategy, trade_g: what the player gets, trade_f: what the player trades
+        """
+        score = 0
+        weight = 0
+        trade_g = None
+        trade_f = None
 
-    def draw_development_card(current_board):
-        return current_board
+        resources = player.resources
+        strategy = player.strategy
+        excess_resources, zero_resources = self.get_excess_and_zeros(resources)
+        lowest_resource = self.get_lowest_resource(resources)
+
+        if strategy == "settlements":
+            if "stone" in excess_resources:
+                trade_f = "stone"
+                weight = 1
+                score = score + 1
+                trade_g = lowest_resource
+
+            elif len(excess_resources) >= 1:
+                weight, score, trade_g, trade_f = self.random_excess_for_lowest(excess_resources, lowest_resource)
+
+        elif strategy == "cities":
+            if "wheat" not in excess_resources and "stone" not in excess_resources:
+                random.shuffle(excess_resources)
+                trade_f = excess_resources.pop(0)
+                weight = 1
+                score = score + 1
+                trade_g = lowest_resource
+
+            elif len(excess_resources) >= 1:
+                weight, score, trade_g, trade_f = self.random_excess_for_lowest(excess_resources, lowest_resource)
+
+        elif strategy == "sheep_monopoly":
```

Comment:
For each of the monopoly strategies you can refactor by creating a new method since they all do the same thing besides the resource that it is a monopoly of.